app/services/preserved_object_handler.rb

```diff
@@ -56,26 +56,35 @@ def initialize(druid, incoming_version, incoming_size, endpoint)
     @endpoint = endpoint
   end
 
+  def create_with_validation
+    results = []
+    if invalid?
+      results << result_hash(INVALID_ARGUMENTS, errors.full_messages)
+    else
+      dt = DruidTools::Druid.new(druid)
+      object_dir = "#{storage_location}/#{dt.tree.join('/')}"
+      moab = Moab::StorageObject.new(druid, object_dir)
+      object_validator = Stanford::StorageObjectValidator.new(moab)
+      validation_errors = object_validator.validation_errors
+      if validation_errors.empty?
+        create
```

Comment:
i'm not sure i understand the call to `create` here, as opposed to a call to `results.concat(create_db_objects(Status.default_status))`... if it's to get the check for an extant `PreservedObject`, wouldn't you want to do that in either case?  entirely possible i'm misunderstanding things, but i'd have expected the two branches of this if to parallel each other a bit more closely.

Reply:
also, i see where result codes and log statements are tested for `create_with_validation` when it encounters validation errors, but not when there are no errors and the creation happens cleanly.

Reply:
good point...:)

Reply:
Also there was a test ``` 'calls #create when there are no validation errors' ```

Reply:
^ but i'm changing it to ` results.concat(create_db_objects(Status.default_status)) `